crates/hc_sandbox/src/generate.rs

```diff
@@ -58,13 +66,70 @@ pub fn generate_with_config(
 pub fn generate_directory(
     root: Option<PathBuf>,
     directory: Option<PathBuf>,
-) -> anyhow::Result<PathBuf> {
+) -> anyhow::Result<(PathBuf, url2::Url2)> {
+    let passphrase = holochain_util::pw::pw_get()?;
+
     let mut dir = root.unwrap_or_else(std::env::temp_dir);
     let directory = directory.unwrap_or_else(|| nanoid::nanoid!().into());
     dir.push(directory);
     std::fs::create_dir(&dir)?;
     let mut keystore_dir = dir.clone();
     keystore_dir.push("keystore");
-    std::fs::create_dir(keystore_dir)?;
-    Ok(dir)
+    std::fs::create_dir(&keystore_dir)?;
+
+    let con_url = init_lair(&keystore_dir, passphrase)?;
+
+    Ok((dir, con_url))
+}
+
+pub(crate) fn init_lair(
+    dir: &std::path::Path,
+    passphrase: sodoken::BufRead,
+) -> anyhow::Result<url2::Url2> {
+    match init_lair_inner(dir, passphrase) {
+        Err(err) => Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!("Failed to execute 'lair-keystore init': {:?}", err),
```

Comment:
instead of coercing this into a single error and stringifying `err` we could make use of https://docs.rs/anyhow/latest/anyhow/trait.Context.html

Reply:
Actually, I'm going to decline to change this. As this is a cli utility, it all just ends up as text on stderr anyways.

Reply:
i see, thanks for explaining